src/lib/logical_query_plan/mock_node.hpp

```diff
@@ -6,28 +6,41 @@
 #include <vector>
 
 #include "abstract_lqp_node.hpp"
+#include "all_type_variant.hpp"
 
 namespace opossum {
 
 /**
- * Node that represents a table that has no data backing it, but may provide (mocked) statistics. Intended for usage in
- * tests (e.g. general LQP tests, optimizer tests that just rely on statistics and not actual data) and the playground
+ * Node that represents a table that has no data backing it, but may provide
+ *  - (mocked) statistics
+ *  - or just a column layout. It will pretend it created the columns.
+ * It is useful in tests (e.g. general LQP tests, optimizer tests that just rely on statistics and not actual data) and
+ * the playground
  */
 class MockNode : public AbstractLQPNode {
  public:
-  MockNode();
-  explicit MockNode(const std::shared_ptr<TableStatistics>& statistics);
+  using ColumnDefinitions = std::vector<std::pair<DataType, std::string>>;
+
+  explicit MockNode(const std::optional<std::string>& alias = std::nullopt);
```

Comment:
When would a mock node without columns make sense?

Reply:
I probably had the idea of building LQPs solely consisting of MockNodes to test basic LQP features but this of course doesn't make sense when nodes have multiple children. Will remove this variant.